### DISCUSSION TITLE
AI Titling

### DISCUSSION THREAD

**Discussion by timnevits** [chhoumann/quickadd#730] [Q&A]
I may have missed it. Is there a way to get the AI Assistant to generate the title of a note, rather than manually coming up with one?

**Comment by chhoumann** [comment]
Very late reply, sorry. Yes - a Macro choice with two steps does this:

1. **AI Assistant command.** Create a note in your prompt templates folder along the lines of:

   ```
   Reply with only a short, filename-safe title for the following:

   {{VALUE:content}}
   ```

   Select it as the command's **Prompt Template**, and set the command's **Output variable name** to `title`.

2. **Your Template choice as the next macro step.** Set its File Name Format to `{{VALUE:title}}`, and use `{{VALUE:content}}` in the template body if you want the captured text in the note too. Variables carry across steps within a macro, so nothing prompts twice.

Running the macro asks you once for the content, sends it to the model, and creates the note under the generated title.

https://quickadd.obsidian.guide/docs/AIAssistant
